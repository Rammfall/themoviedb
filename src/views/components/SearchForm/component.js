import React from 'react'
import { Col, Row } from 'antd'
import { CloseOutlined } from '@ant-design/icons'
import { Form, Field } from 'formik'
import PropTypes from 'prop-types'

import InputField from 'Views/components/shared/InputField'

const SearchFormComponent = ({ handleSubmit, isLoading, handleReset }) => (
  <Row
    justify='center'
  >
    <Col
      className='gutter-row'
      xs={{ span: 20 }}
      sm={{ span: 20 }}
      md={{ span: 14 }}
      lg={{ span: 12 }}
      xl={{ span: 10 }}
    >
      <Form>
        <Field
          name='search'
          component={InputField}
          type='search'
          placeholder='Enter movie name'
          size='large'
          enterButton='Search'
          className='top-margin'
          prefix={<CloseOutlined onClick={handleReset} />}
          onSearch={handleSubmit}
          loading={isLoading}
        />
      </Form>
    </Col>
  </Row>
)

SearchFormComponent.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

SearchFormComponent.defaultProps = {
  isLoading: false
}

export default SearchFormComponent
