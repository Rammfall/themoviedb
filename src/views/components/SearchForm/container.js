import React from 'react'
import { withFormik } from 'formik'
import { connect } from 'react-redux'
import { compose } from 'ramda'
import * as yup from 'yup'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import queryParams from 'Utils/router/queryParams'
import searchSubmit from 'Utils/formikEvents/searchSubmit'
import { loadingSelector } from 'Store/concepts/data/selectors'
import { dashboardConstant } from 'Constants/concepts'

import SearchFormComponent from './component'

const SearchForm = ({ handleSubmit, isLoading, handleReset }) => (
  <SearchFormComponent
    handleSubmit={handleSubmit}
    isLoading={isLoading}
    handleReset={handleReset}
  />
)

SearchForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

SearchForm.defaultProps = {
  isLoading: false
}

const mapStateToProps = (state) => ({
  isLoading: loadingSelector(state, dashboardConstant)
})

export { SearchForm as SearchFormContainer }
export default compose(
  withRouter,
  connect(mapStateToProps),
  withFormik({
    validationSchema: yup.object().shape({
      search: yup.string()
    }),
    mapPropsToValues: ({ location: { search: query } }) => {
      const searchQuery = decodeURIComponent(queryParams(query).get('search') || '')

      return {
        search: searchQuery
      }
    },
    handleSubmit: searchSubmit
  })
)(SearchForm)
