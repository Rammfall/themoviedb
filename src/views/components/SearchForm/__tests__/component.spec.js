import React from 'react'
import { shallow } from 'enzyme'

import SearchForm from '../component'

describe('SearchForm', () => {
  it('matches snapshot', () => {
    const component = shallow(<SearchForm />)

    expect(component).toMatchSnapshot()
  })
})
