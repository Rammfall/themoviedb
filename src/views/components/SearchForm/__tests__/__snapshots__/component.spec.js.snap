// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFormComponent matches snapshot 1`] = `
<Row
  justify="center"
>
  <Col
    className="gutter-row"
    lg={
      Object {
        "span": 12,
      }
    }
    md={
      Object {
        "span": 14,
      }
    }
    sm={
      Object {
        "span": 20,
      }
    }
    xl={
      Object {
        "span": 10,
      }
    }
    xs={
      Object {
        "span": 20,
      }
    }
  >
    <Form>
      <Field
        className="top-margin"
        component={[Function]}
        enterButton="Search"
        loading={false}
        name="search"
        onSearch={[Function]}
        placeholder="Enter movie name"
        prefix={
          <ForwardRef(CloseOutlined)
            onClick={[Function]}
          />
        }
        size="large"
        type="search"
      />
    </Form>
  </Col>
</Row>
`;
